### PR TITLE
Find Currency by numeric code

### PR DESCRIPTION
**Pull request by Stocco**
Hello, I'm adding the option to search Currencies by its numeric code ISO number. 
This helps because many implementations works around the iso numerical code numbers (like Visa and Mastercard for instance).

-  This code will load a second map of currencies but now this map uses the numerical ISO code as key of the map.
-  The reason to have another map is to avoid having a loop each time we need to access a currency its iso numerical code
- Added idea and ds_store files to git ignore

### CODE REVIEW

**Comment by Stocco** [comment]
@Rhymond Can you take a look at it, please?

**Comment by rodrigopandini** [comment]
@Rhymond Would be nice check this PR.

File: currency.go [hunk 1]
```diff
@@ -15,177 +16,194 @@ type Currency struct {
...
-	XCD: {Decimal: ".", Thousand: ",", Code: XCD, Fraction: 2, NumericCode: "951", Grapheme: "$", Template: "$1"},
-	XDR: {Decimal: ".", Thousand: ",", Code: XDR, Fraction: 0, NumericCode: "960", Grapheme: "SDR", Template: "1 $"},
-	XPF: {Decimal: ".", Thousand: ",", Code: XPF, Fraction: 0, NumericCode: "953", Grapheme: "₣", Template: "1 $"},
-	YER: {Decimal: ".", Thousand: ",", Code: YER, Fraction: 2, NumericCode: "886", Grapheme: "\ufdfc", Template: "1 $"},
-	ZAR: {Decimal: ".", Thousand: ",", Code: ZAR, Fraction: 2, NumericCode: "710", Grapheme: "R", Template: "$1"},
-	ZMW: {Decimal: ".", Thousand: ",", Code: ZMW, Fraction: 2, NumericCode: "967", Grapheme: "ZK", Template: "$1"},
-	ZWD: {Decimal: ".", Thousand: ",", Code: ZWD, Fraction: 2, NumericCode: "932", Grapheme: "Z$", Template: "$1"},
+var (
+	// ErrCurrencyNotFound happens when it wasn't possible to find the currency by the given parameter.
+	ErrCurrencyNotFound = errors.New("currency not found")
```
Comment (Rhymond) [inline currency.go:21]: error is not used, let's remove it

File: currency.go [hunk 2]
```diff
@@ -15,177 +16,194 @@ type Currency struct {
...
+		YER: {Decimal: ".", Thousand: ",", Code: YER, Fraction: 2, NumericCode: "886", Grapheme: "\ufdfc", Template: "1 $"},
+		ZAR: {Decimal: ".", Thousand: ",", Code: ZAR, Fraction: 2, NumericCode: "710", Grapheme: "R", Template: "$1"},
+		ZMW: {Decimal: ".", Thousand: ",", Code: ZMW, Fraction: 2, NumericCode: "967", Grapheme: "ZK", Template: "$1"},
+		ZWD: {Decimal: ".", Thousand: ",", Code: ZWD, Fraction: 2, NumericCode: "932", Grapheme: "Z$", Template: "$1"},
+	}
+	currenciesByNumericCode = loadNumericCodeCurrenciesMap()
+)
+
+// loadNumericCodeCurrenciesMap adds a second map with its key value being the numeric ISO code.
+func loadNumericCodeCurrenciesMap() (currenciesByCode map[string]*Currency) {
```
Comment (Rhymond) [inline currency.go:199]: This is not going to work with the custom currencies added with `AddCurrency`. To achieve the same effect. we must iterate through the whole map each time

File: currency.go [hunk 3]
```diff
@@ -15,177 +16,194 @@ type Currency struct {
...
+	currenciesByCode = make(map[string]*Currency)
```
Comment (Rhymond) [inline currency.go:200]: you know the size of the map

```suggestion
	currenciesByCode = make(map[string]*Currency, len(currencies))
```

File: currency_test.go [hunk 4]
```diff
@@ -88,3 +88,38 @@ func TestCurrency_GetNonExistingCurrency(t *testing.T) {
...
+			"happy-currency-find",
+			args{code: "986"},
+			&Currency{Decimal: ",", Thousand: ".", Code: BRL, Fraction: 2, NumericCode: "986", Grapheme: "R$", Template: "$1"},
+		},
+		{
+			"happy-currency-not-found",
+			args{code: "1111"},
+			nil,
+		},
+		{
```
Comment (Rhymond) [inline currency_test.go:111]: Let's add a test case with added currency with `AddCurrency`

**Comment by Rhymond** [comment]
The feature was released in v1.0.5. Thanks All. Closing

**Pull request closed without merging**